chimneyCats/src/test/scala/io/scalaland/chimney/cats/CatsValidatedSpec.scala

```diff
@@ -442,6 +442,72 @@ object CatsValidatedSpec extends TestSuite {
           .intoF[ValidatedNec[String, +*], long.NumScale[Int]]
           .transform ==> Validated.invalidNec("bad int")
       }
+
+      "path of error" - {
+        type V[+A] = ValidatedNec[TransformationError[String], A]
+
+        def printError(err: TransformationError[String]): String =
+          s"${err.message} on ${err.showErrorPath}"
+
+        implicit val intParse: TransformerF[V, String, Int] =
+          str =>
+            Validated.fromOption(
+              str.parseInt,
+              NonEmptyChain.one(TransformationError[String](s"Can't parse int from $str"))
+            )
+
+        "case classes" - {
+          case class Foo(a: String, b: String, c: InnerFoo)
+
+          case class InnerFoo(d: String, e: String)
+
+          case class Bar(a: Int, b: Int, c: InnerBar)
+
+          case class InnerBar(d: Int, e: Int)
+
+          Foo("mmm", "nnn", InnerFoo("lll", "jjj")).transformIntoF[V, Bar].leftMap(_.map(printError)) ==>
+            Validated.Invalid(
+              NonEmptyChain(
+                "Can't parse int from mmm on a",
+                "Can't parse int from nnn on b",
+                "Can't parse int from lll on c.d",
+                "Can't parse int from jjj on c.e"
+              )
+            )
+        }
+
+        "list" - {
+          case class Foo(list: List[String])
+
+          case class Bar(list: List[Int])
+
+          Foo(List("a", "b", "c")).transformIntoF[V, Bar].leftMap(_.map(printError)) ==>
+            Validated.Invalid(
+              NonEmptyChain(
+                "Can't parse int from a on list[0]",
+                "Can't parse int from b on list[1]",
+                "Can't parse int from c on list[2]"
+              )
+            )
+        }
+
+        "map" - {
+          case class Foo(map: Map[String, String], map2: Map[String, String])
+
+          case class Bar(map: Map[Int, Int], map2: Map[String, Int])
+
+          Foo(Map("a" -> "b", "c" -> "d"), Map("e" -> "f")).transformIntoF[V, Bar].leftMap(_.map(printError)) ==>
+            Validated.Invalid(
+              NonEmptyChain(
+                "Can't parse int from a on map.keys",
```

Comment:
I curious about few more cases:

1. What would be the path in case that we have case class as map keys? I believe for `case class CCKey(value: String)` it should be printed as `map[CCKey(a)]`, shouldn't it?

2. I see that you implemented path reporting for case where map key can't be transformed. That's very nice 👍 But will it always end on `map.keys`? What if we have some advanced transformation in map keys that also carry some path info? Would it be composed somehow?

Reply:
1. Added test
2. Value of key is able in transformer and directly adding to error message, for example in my case message `"Can't parse int from a on map.keys"` contains information in what case it failed

Reply:
1. 👍 
2. The information comes from a message provided by user in error case of function passed to `withFieldComputedF`, not from a path itself. The path here informs us that the problem relates to some key, but we don't know which one (just by looking at the path). Let me elaborate on more advanced example:

```scala
case class KeyFoo(innerField: String)
case class KeyBar(innerField: Int)

case class Foo(map: Map[KeyFoo, Boolean])
case class Bar(map: Map[KeyBar, Boolean])
```

Having implicit `TransformerF[..., String, Int]` and trying to transform:

```scala
Foo(Map(KeyFoo("nan") -> True)).transformIntoF[..., Bar]
```

...my question was about what error path we obtain in this case?

Note that `map.keys` path doesn't really inform us which key the error regards. From the inner transformation of `KeyFoo` to `KeyBar` we can extract path information so that it failed because `innerField` field can't be parsed to int.

Do you think we can somehow enrich the path so that it carry both information?
In this case it should be something similar to `map.KeyFoo(nan).innerField` (I'm not stubborn about the syntax).

Reply:
Fixed